### ISSUE TITLE
Cannot round trip empty arrays

### ISSUE THREAD

**Issue by GPSnoopy** [G-Research/ParquetSharp#47]
Trying to round trip empty arrays does not currently work, as our array handling logic is not bullet proof.

**Comment by philjdf** [comment]
A column which reproduces this is:
```csharp
/*
* [None, [], [1.0, None, 2.0]]
* []
* None
* [[]]
*/
var expected = new double?[][][] {
    new double?[][] { null, new double?[] { }, new double?[] {1.0, null, 2.0} },
    new double?[][] {},
    null,
    new double?[][] { new double?[] { } }
};
```

It should read/write the following:

```
values = [1.0, 2.0]
def_levels = [2, 3, 5, 4, 5, 1, 0, 3]
rep_levels = [0, 1, 1, 2, 2, 0, 0, 0]
```

**Comment by GPSnoopy** [comment]
Fixed in 1.5.1.1 beta 5
